### PR TITLE
optimize:  simplify if else judgment

### PR DESCRIPTION
**Pull request by coconut4j**
### Ⅰ. Describe what this PR did
use two ImmutableMap instead of lots of if else judge, and the code is more readability,more convenient to modify.

### Ⅱ. Does this pull request fix one issue?

切换develop分支 以及将map并入构造器   旧pr #3932

### CODE REVIEW

**Comment by caohdgege** [comment]
can the DatabaseTransactionStoreManager be optimize like this?

**Comment by coconut4j** [comment]
> can the DatabaseTransactionStoreManager be optimize like this?
DataBaseTransactionStoreManager can use one map to solve it

**Review by l81893521 — APPROVED** [review]
LGTM

**Review by l81893521 — COMMENTED** [review]
Can you put some change log in change module?

**Comment by coconut4j** [comment]
> Can you put some change log in change module like other PR?

ok，i finished it

**Review by a364176773 — APPROVED** [review]
LGTM

File: changes/en-us/1.5.0.md [hunk 1]
```diff
@@ -89,7 +89,7 @@
   - [[#3916](https://github.com/seata/seata/pull/3916)] optimize determine whether the server in the register is alive
   - [[#3918](https://github.com/seata/seata/pull/3918)] cache reflection results of the fields and methods
   - [[#3912](https://github.com/seata/seata/pull/3912)] support config JVM param in env
-
+  - [[#3912](https://github.com/seata/seata/pull/3939)] use map instead of if else judge for more change in the future
```
Comment (l81893521) [inline changes/en-us/1.5.0.md:92]: #3939

Comment (coconut4j) [reply changes/en-us/1.5.0.md]: > #3939

sorry, it fixed it now

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java [hunk 2]
```diff
@@ -94,20 +96,65 @@ public static RedisTransactionStoreManager getInstance() {
...
-        } else if (LogOperation.BRANCH_ADD.equals(logOperation)) {
-            return insertBranchTransactionDO(SessionConverter.convertBranchTransactionDO(session));
-        } else if (LogOperation.BRANCH_UPDATE.equals(logOperation)) {
-            return updateBranchTransactionDO(SessionConverter.convertBranchTransactionDO(session));
-        } else if (LogOperation.BRANCH_REMOVE.equals(logOperation)) {
-            return deleteBranchTransactionDO(SessionConverter.convertBranchTransactionDO(session));
+        if (globalMap.containsKey(logOperation) || branchMap.containsKey(logOperation)) {
+            return globalMap.containsKey(logOperation) ?
+                    globalMap.get(logOperation).apply(SessionConverter.convertGlobalTransactionDO(session)) :
+                    branchMap.get(logOperation).apply(SessionConverter.convertBranchTransactionDO(session));
```
Comment (drgnchan) [inline server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:157]: In my opinion,there is no need to invoke `globalMap.containsKey(logOperation)` again.it is better to break the `if` statement into two `if` statements.

**Review by a364176773 — CHANGES_REQUESTED** [review]
为何不把db那块实现一并修改?

File: server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java (hunk 2 above)
Comment (a364176773) [reply server/src/main/java/io/seata/server/storage/redis/store/RedisTransactionStoreManager.java:157]: +1

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**